Add a bunch of new notification icons

jplexer opened the pull request:
Specifically: Beeper, Bluesky, Discord, Duolingo, Element, Google Chat, Google Tasks, Home Assistant, Steam, Teams, Threads, UniFi Protect, Zoom, and eBay

hellcp committed:
tools/generate_pdcs: Implement group/layer inheritence when converting from svg

Signed-off-by: Stasia Michalska <hel@lcp.world>
Signed-off-by: Joshua Jun <joshuajun@proton.me>

jplexer committed:
third_party: add iconography repo

Signed-off-by: Joshua Jun <joshuajun@proton.me>

jplexer committed:
tools/generate_pdcs: fix parsing of space-separated polygon points

The get_points_from_str function assumed points were always in "x,y"
comma-separated format, but some SVG files use space-separated
coordinates like "x1 y1 x2 y2". This caused an IndexError when parsing
polygons or polylines with this format.

Fix by normalizing the point string (replacing commas with spaces) and
processing values in pairs, which handles all common SVG formats.

Signed-off-by: Joshua Jun <joshuajun@proton.me>

jplexer committed:
tools/resources: support paths outside resources directory

Allow resource definitions to reference files outside the resources/
directory using relative paths (e.g. ../third_party/...). Previously,
all paths were passed through find_most_specific_filename which only
searched within the resources directory.

Paths starting with ".." are now normalized with os.path.normpath()
instead, enabling resources to be sourced from third_party repos like
the iconography submodule.

Signed-off-by: Joshua Jun <joshuajun@proton.me>

jplexer committed:
resources: add more notification icons

Signed-off-by: Joshua Jun <joshuajun@proton.me>

jplexer committed:
fw/ancs: add known app mappings for new notification icons

Signed-off-by: Joshua Jun <joshuajun@proton.me>

jplexer committed:
resources: use newer outlook logo

Signed-off-by: Joshua Jun <joshuajun@proton.me>

jplexer committed:
fw/resource: Increase system resource limit to 768

Signed-off-by: Joshua Jun <joshuajun@proton.me>

hellcp commented on resources/normal/base/resource_map.json:
```diff
@@ -3080,6 +3290,132 @@
+      "sizes": {
+        "tiny": "RESOURCE_ID_NOTIFICATION_GOOGLE_TASKS_TINY",
+        "small": "RESOURCE_ID_NOTIFICATION_GOOGLE_TASKS_SMALL",
+        "large": "RESOURCE_ID_NOTIFICATION_GOOGLE_TASKS_LARGE"
```
This seems like not the worst thing to replace google inbox with, unless you really do not want to mess with replacing old ids

jplexer committed:
tools/pdc_gen: fix Python 3 bytes/string compatibility

Change output initialization from empty string to empty bytes literal.
The serialize functions return bytes, so the initial value must also be
bytes to avoid TypeError when the output is used in pbpack.

Signed-off-by: Joshua Jun <joshuajun@proton.me>

ericmigi reviewed (approved):
looks great!

lavglaab commented:
youtube icons do exist in https://github.com/pebble-dev/iconography ! is there a reason they are not included in this PR?

jplexer commented:
@ericmigi #572
